fiftyone/core/dataset.py

```diff
@@ -2293,27 +2295,91 @@ def save_view(self, name, view, overwrite=False):
         if view._root_dataset is not self:
             raise ValueError("Cannot save view into a different dataset")
 
-        if name in self._doc.views:
+        if self.has_view(name):
             if not overwrite:
                 raise ValueError(
                     "Saved view with name '%s' already exists" % name
                 )
 
-            _view_doc = self._doc.views.pop(name)
+            _view_doc = self._get_view_doc(name, pop=True)
             _view_doc.delete()
 
+        now = datetime.utcnow()
         view_doc = foo.ViewDocument(
             dataset_id=self._doc.id,
+            name=name,
+            url_name=fou.to_url_name(name),
+            description=description,
+            color=color,
             view_stages=[
                 json_util.dumps(s)
                 for s in view._serialize(include_uuids=False)
             ],
+            index=len(self._doc.views),
```

Comment:
what does the index do?

Reply:
`index` is so the views can be ordered. The figma design allows them to be reordered so that you can have your favorite view listed first, for example.

I added `_ordered_view()` and `_reorder_views()` as private methods to demonstrate how `index` is used. Also added some unit tests for this functionality.

LMK if you have any thoughts on a better way to implement "views can be reordered"

Reply:
makes sense 👍

Reply:
FYI I removed `index` for now and am instead relying on the ordering of the `DatasetDocument.views` list to convey this information